velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -466,6 +470,32 @@ struct TimestampMinusIntervalDayTime {
   }
 };
 
+template <typename T>
+struct TimestampAtTimezoneFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
```

Comment:
can we optimize for the case where the timezone string is constant (which is probably the vast majority of cases)? For an example, check `DateTruncFunction`

Reply:
I checked DateTrunc but don't quite understand what you mean, could you elaborate?

Reply:
you can provide an init() method in your class, and check there if the function received any constant parameters. In case the timezone string supplied in a constant value, you can performance the conversion just once per batch, not once per row.